### DISCUSSION TITLE
How to pass an array of dates to selectedDays?

### DISCUSSION THREAD

**Discussion by MattHeslington** [gpbl/react-day-picker#1178] [Support] [answered]
Hello,

I'm using react-day-picker to display a calendar in which I would like to show all the dates of posts on a 'blog'. Users then click on a day to view that day's post. It works very well if I pass the dates in manually (dateArray is an array of dates):

```jsx
<DayPicker
            onDayClick={handleDayClick}
            selectedDays={[
                new Date(dateArray[0]),
                new Date(dateArray[1]),
                new Date(dateArray[2]),
                etc....
            ]}
        />
```

However, is it possible to just pass in the 'dateArray' without the [0], [1] etc. so as to make it truly dynamic?

Many thanks for any help,
Matt

**Comment by gpbl** [comment, marked as answer]
@MattHeslington I suspect that dateArray contains strings? What is its content? You can also write:

```
selectedDays={ dateArray.map(date => newDate(date)) }
```

**Reply by MattHeslington** [reply]
Fantastic Giampaolo, that works perfectly. Many, many thanks. (Yes, my dateArray did contain strings)
